### PR TITLE
MON-3763: Add cnv_abnormal

### PR DESCRIPTION
**Pull request by avlitman**
Add new recording rule about cnv issues to telemeter, as for now the only issue is memory exceeded.

* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml [hunk 1]
```diff
@@ -240,6 +240,11 @@ data:
     #
     # owners: (https://github.com/kubevirt)
     #
+    # cnv_abnormal holds issues with the pods for each container, e.g. memory exceeded value.
```
Comment (sradco) [inline manifests/0000_50_cluster-monitoring-operator_04-config.yaml:243]: @avlitman 
    # cnv_abnormal represents the reason why the operator might have an issue
    # and includes the node, namespace, container, reason labels.

Comment (avlitman) [reply manifests/0000_50_cluster-monitoring-operator_04-config.yaml]: fixed

**Comment by avlitman** [comment]
/unhold

Created created https://issues.redhat.com/browse/MON-3763
And the pr adding this metric is merged as well: https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2855

@jan--f Will appreciate your review and approval, also should I cherry-pick this to 4.16? since I see we are already have 4.17. thanks

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml [hunk 2]
```diff
@@ -240,6 +240,12 @@ data:
     #
     # owners: (https://github.com/kubevirt)
     #
+    # cnv_abnormal represents the reason why the operator might have an issue
+    # and includes the node, namespace, container, reason labels.
```
Comment (simonpasquier) [inline manifests/0000_50_cluster-monitoring-operator_04-config.yaml:244]: Why include the node name? Is it meaningful?

**Review by simonpasquier — COMMENTED** [review]
Looking at https://github.com/kubevirt/hyperconverged-cluster-operator/blob/a98cb26b9ad8774cd676f703600bb0fe407875cc/pkg/monitoring/rules/recordingrules/operator.go#L26-L33, I'm not sure why you have a recording rule only renaming the metric name. Why not sending the original metric?

For reference:
* https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2855
* https://github.com/kubevirt/kubevirt/pull/11557

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml (hunk 2 above)
Comment (simonpasquier) [reply manifests/0000_50_cluster-monitoring-operator_04-config.yaml:244]: the `node` label wasn't mentioned in https://issues.redhat.com/browse/MON-3763

**Comment by simonpasquier** [comment]
> also should I cherry-pick this to 4.16? since I see we are already have 4.17.

For now 4.16 == 4.17 (until the 4.16 branch is officially cut).

**Comment by simonpasquier** [comment]
you'll need to run `make generate` and commit the changes too.

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml (hunk 2 above)
Comment (avlitman) [reply manifests/0000_50_cluster-monitoring-operator_04-config.yaml:244]: Yes,, we need to node label for sure.

Comment (avlitman) [reply manifests/0000_50_cluster-monitoring-operator_04-config.yaml:244]: but the namespace needs to be removed

**Comment by avlitman** [comment]
> make generate

@simonpasquier done (:

**Comment by avlitman** [comment]
@simonpasquier @jan--f this pr is ready to be merged, we remove the node: 
https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2945
https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2955
unfortunately they didn't agree to change the name but there not gonna be many reasons.

**Comment by avlitman** [comment]
@simonpasquier @jan--f this pr is ready to be merged, we remove the node:
https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2945
https://github.com/kubevirt/hyperconverged-cluster-operator/pull/2955
unfortunately they didn't agree to change the name but there not gonna be many reasons.

**Comment by avlitman** [comment]
Thanks @jan--f. Just want to make sure- should I cherry pick this to 4.16 or the main is still 4.16?

**Comment by avlitman** [comment]
/retitle [OCPBUGS-34805](https://issues.redhat.com//browse/OCPBUGS-34805): Add cnv_abnormal

**Pull request merged**